commit 750946c8bfce458490e7e4aba34f22893c16a297
Author: J. R. Okajima <hooanon05g@gmail.com>
Date:   Mon Jan 23 03:36:33 2017 +0900

    aufs: for v4.9, support posix acl
    
    The commit
    	3801e27 2016-12-08 aufs: for v4.9-rc1, support new xattr scheme
    was not enough, and supporting posix acl was missing.
    
    Reported-by: Justin Cormack <justin.cormack@docker.com>
    See-also: https://github.com/docker/docker/issues/30245#issuecomment-274152452
    Signed-off-by: J. R. Okajima <hooanon05g@gmail.com>

diff --git a/fs/aufs/xattr.c b/fs/aufs/xattr.c
index d0834e4..35fcc50 100644
--- a/fs/aufs/xattr.c
+++ b/fs/aufs/xattr.c
@@ -19,6 +19,8 @@
  * handling xattr functions
  */
 
+#include <linux/fs.h>
+#include <linux/posix_acl_xattr.h>
 #include <linux/xattr.h>
 #include "aufs.h"
 
@@ -322,7 +324,11 @@ static const struct xattr_handler au_xattr_handler = {
 };
 
 static const struct xattr_handler *au_xattr_handlers[] = {
-	&au_xattr_handler,
+#ifdef CONFIG_FS_POSIX_ACL
+	&posix_acl_access_xattr_handler,
+	&posix_acl_default_xattr_handler,
+#endif
+	&au_xattr_handler, /* must be last */
 	NULL
 };
 
